databrowser/src/layouts/menu/OverlayMenu.vue

```diff
@@ -0,0 +1,131 @@
+<template>
+  <button
+    v-if="dialogOpen"
+    :title="$t('header.menu.closeAction')"
+    class="md:hidden"
+    @click="closeDialog"
+  >
+    <IconClose />
+  </button>
+  <button
+    v-else
+    :title="$t('header.menu.openAction')"
+    class="inline-flex space-x-4"
+    @click="dialogOpen = true"
+  >
+    <IconMenu />
+    <span class="hidden md:block">{{ $t('header.menu.title') }}</span>
+  </button>
+
+  <Dialog
+    :open="dialogOpen"
+    class="fixed top-0 z-20 p-0 w-screen"
+    @close="closeDialog"
+  >
+    <DialogOverlay class="hidden md:block fixed inset-0 bg-black opacity-30" />
+    <!-- Desktop Menu -->
+    <div
+      class="
+        hidden
+        md:block
+        overflow-x-auto
+        absolute
+        top-0
+        left-0
+        h-screen
+        bg-white
+      "
+    >
+      <button class="block py-4 px-4" @click="closeDialog">
+        <IconClose />
+      </button>
+      <div class="inline-flex flex-row h-full divide-x-2">
+        <OveralyMenuList
+          v-for="(item, index) in navigation"
+          :key="index"
+          :item="item"
+          @select-category="(menu) => addSubMenu(menu, index)"
+        />
+      </div>
+    </div>
+
+    <!-- Mobile Menu -->
+    <div
+      class="
+        md:hidden
+        absolute
+        top-12
+        right-0
+        left-0
+        px-4
+        h-screen
+        bg-white
+        border-t
+      "
+    >
+      <OveralyMenuList
+        :item="lastElement"
+        :show-arrow="navigation.length !== 1"
+        @select-category="(menu) => navigation.push(menu)"
```

Comment:
Just a question here because I'm curious. Is it here not also possible to use the `addSubMenu` function?

Reply:
Since addSubMenu also has an index and if this is not available in the mobile view.

I have now simply changed the method a bit so that the index is optional

Reply:
Sounds good. It was just a quick question 😉